crates/sui-rest-api/src/accounts.rs

```diff
@@ -55,13 +55,13 @@ async fn list_account_objects(
     let mut object_info = state
         .inner()
         .account_owned_objects_info_iter(address.into(), start)?
+        .take(limit + 1)
         .map(|info| AccountOwnedObjectInfo {
             owner: info.owner.into(),
             object_id: info.object_id.into(),
             version: info.version.into(),
-            type_: struct_tag_core_to_sdk(info.type_.into()),
+            type_: struct_tag_core_to_sdk(info.type_.into()).expect("object types must be valid"),
```

Comment:
They should, but do we need to panic if they're not, or could we return an internal error instead?